sharepa/analysis.py

```diff
@@ -16,6 +16,33 @@ def bucket_to_dataframe(name, bucket):
     ).T
 
 
+def convert_key(x):
+    x.key = x.key_as_string
+    return x
+
+
+def aggregation_to_dataframe(name,agg,filter=None):
+    '''A function that turns elasticsearch response with/without aggregation into a dataframe
+
+        :param name: The name of the returned dataframe col
+        :type name: str
+        :param bucket: a bucket from elasticsearch results
+        :type bucket: list[dict]
+        :returns: pandas.DataFrame
+    '''
+    if hasattr(agg, 'buckets'):
+        if filter is not None:
+            expanded_agg = []
+            for bucket in agg.buckets:
+                expanded_agg.append(bucket_to_dataframe(bucket.key, map(convert_key, getattr(bucket,filter).buckets)))
```

Comment:
Would have to see how this would affect plotting/merging though. May complicate those.